### PR TITLE
fix: navbar user

### PR DESCRIPTION
**Pull request by hermannm**
## Proposed Changes

- (hopefully) fix edge case mis-styling of user button in navbar
- adjust navbar breakpoint to reflect updated navbar content
- refactor navbar styles
  - move to using theme.spacing more than hardcoded values

## Types of Changes

- [ ] New feature
- [x] Bug fix
- [ ] Enhancement/optimization
- [ ] Refactor
- [x] Style
- [ ] Build/dependencies
- [ ] Other

## Checklist

- [ ] I have added tests to cover my changes (for features/bugs)
- [x] I am pleased with the readability of the code (if not, state where you need input)
- [x] I have tested the changes and verified that they work and do not break anything (to the best of my ability)

### CODE REVIEW

File: frontend/src/components/navbar/NavbarLinks/index.tsx [hunk 1]
```diff
@@ -1,65 +1,27 @@
...
+      {links.map(
+        (link) =>
+          (!link.loginRequired || (link.loginRequired && loggedIn)) &&
+          (link.permissionRequired ? (
+            <PermissionRequired permission={link.permissionRequired}>
+              <NavbarItem link={link} />
+            </PermissionRequired>
+          ) : (
+            <NavbarItem link={link} />
+          ))
```
Comment (larwaa) [inline frontend/src/components/navbar/NavbarLinks/index.tsx:22]: This should function the same. If logged in, show regardless of loginRequired. If not, only show those without loginRequired.
```suggestion
          (loggedIn || !link.loginRequired) &&
          (link.permissionRequired ? (
            <PermissionRequired permission={link.permissionRequired}>
              <NavbarItem link={link} />
            </PermissionRequired>
          ) : (
            <NavbarItem link={link} />
          ))
```

**Review by larwaa — CHANGES_REQUESTED** [review]

File: frontend/src/components/navbar/NavbarLinks/index.tsx (hunk 1 above)
Comment (larwaa) [inline frontend/src/components/navbar/NavbarLinks/index.tsx:22]:
```suggestion
      {links.filter((link) => loggedIn || !link.loginRequired).map((link) =>
          (link.permissionRequired ? (
            <PermissionRequired permission={link.permissionRequired}>
              <NavbarItem link={link} />
            </PermissionRequired>
          ) : (
            <NavbarItem link={link} />
          ))
```

Comment (hermannm) [reply frontend/src/components/navbar/NavbarLinks/index.tsx]: Good catch! Fixed in 0ca67f1.

**Review by larwaa — DISMISSED** [review]
_LoOkS_ gOoD **tO** mE

File: frontend/src/components/navbar/NavbarLinks/index.tsx (hunk 1 above)
Comment (hermannm) [reply frontend/src/components/navbar/NavbarLinks/index.tsx]: Good suggestion! Added in 5b85153.

**Review by larwaa — APPROVED** [review]
LooKs GREAT to Me

**Pull request merged**